hummingbot/connector/exchange/bittrex/bittrex_exchange.py

```diff
@@ -0,0 +1,403 @@
+            fee = TradeFeeBase.new_spot_fee(
+                fee_schema=self.trade_fee_schema(),
+                trade_type=order.trade_type,
+                percent_token=percent_token,
+                flat_fees=[TokenAmount(amount=Decimal(trade["commission"]), token=percent_token)]
+            )
+            trade_update = TradeUpdate(
+                trade_id=str(trade["id"]),
+                client_order_id=order.client_order_id,
+                exchange_order_id=exchange_order_id,
+                trading_pair=order.trading_pair,
+                fee=fee,
+                fill_base_amount=Decimal(trade["quantity"]),
+                fill_quote_amount=Decimal(trade["quantity"]) * Decimal(trade["rate"]),
+                fill_price=Decimal(trade["rate"]),
+                fill_timestamp=self._get_timestamp(trade["executedAt"]),
+            )
+            trade_updates.append(trade_update)
+
+        return trade_updates
+
+    async def _request_order_status(self, tracked_order: InFlightOrder) -> OrderUpdate:
+        exchange_order_id = await tracked_order.get_exchange_order_id()
+        params = {
+            "orderId": exchange_order_id
+        }
+        order_update = await self._api_get(
+            path_url=CONSTANTS.ORDER_DETAIL_URL.format(exchange_order_id),
+            params=params,
+            is_auth_required=True,
+            limit_id=CONSTANTS.ORDER_DETAIL_LIMIT_ID
+        )
+        new_state = self._get_order_status(order_update)
+        update_time = self._get_timestamp(order_update["updatedAt"])
+        update = OrderUpdate(
+            client_order_id=tracked_order.client_order_id,
+            exchange_order_id=exchange_order_id,
+            trading_pair=tracked_order.trading_pair,
+            update_timestamp=update_time,
+            new_state=new_state,
+        )
+        return update
+
+    async def _user_stream_event_listener(self):
+        async for stream_message in self._iter_user_event_queue():
+            try:
+                content = stream_message.get("delta") or stream_message.get("deltas")
+                if isinstance(content, List):
+                    safe_ensure_future(self._process_execution_event(content))
+                elif "marketSymbol" not in content:
+                    asset_name = content["currencySymbol"]
+                    total_balance = content["total"]
+                    available_balance = content["available"]
+                    self._account_available_balances[asset_name] = available_balance
+                    self._account_balances[asset_name] = total_balance
+                else:
+                    safe_ensure_future(self._process_order_update_event(content))
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().error("Unexpected error in user stream listener loop.", exc_info=True)
+                await self._sleep(5.0)
```

Comment:
I dont believe we should need a sleep here.

Reply:
Removed the _sleep()